conduit/pipeline/pipeline.go

```diff
@@ -201,17 +225,95 @@ func (p *pipelineImpl) registerPluginMetricsCallbacks() {
 	}
 }
 
-func (p *pipelineImpl) makeConfig(pluginType, pluginName string, cfg []byte) (config plugins.PluginConfig) {
-	config.Config = string(cfg)
+func (p *pipelineImpl) makeConfig(cfg NameConfigPair, pluginType plugins.PluginType) (*log.Logger, plugins.PluginConfig, error) {
```

Comment:
Expanded to initialize the logger and marshal the config file in addition to creating the plugin's data directory.